tests/interfaces-network/task.yaml

```diff
@@ -0,0 +1,85 @@
+summary: |
+  The network interface allows a snap to access the network as a client.
+
+  A snap which defines the network plug must be shown in the interfaces list.
+  The plug must be autoconnected on install and, as usual, must be able to be
+  reconnected.
+
+  A snap declaring a plug on this interface must be able to access network services.
+
+environment:
+  SNAP_NAME: network-consumer
+  SNAP_FILE: "./$[SNAP_NAME]_1.0_all.snap"
+  PLUG: network
+  PORT: 8081
+  REQUEST_FILE: "./request.txt"
+  SERVICE_FILE: "./service.sh"
+  SERVICE_NAME: "test-service"
+
+kill-wait: 1m
+
+prepare: |
+  echo "Given a snap declaring the $PLUG plug is installed"
+  snapbuild ../lib/snaps/$SNAP_NAME .
+  sudo snap install $SNAP_FILE
+
+  echo "And a service is listening"
+  echo "#!/bin/sh\nwhile true; do echo \"HTTP/1.1 200 OK\n\nok\n\" |  nc -l -p $PORT -q 1; done" > $SERVICE_FILE
+  chmod a+x $SERVICE_FILE
+  sudo systemd-run --unit $SERVICE_NAME $SERVICE_FILE
+  while ! netstat -lnt | grep -Pq "tcp.*?:$PORT +.*?LISTEN\n*"; do sleep 0.5; done
+
+  echo "And we store a basic HTTP request"
+  cat > $REQUEST_FILE <<EOF
+  GET / HTTP/1.0
+
+  EOF
+
+restore: |
+  sudo snap remove $SNAP_NAME
+  rm -f $SNAP_FILE $REQUEST_FILE
+  sudo systemctl stop $SERVICE_NAME
```

Comment:
It's a good practice to first stop the activity, then remove content. Not sure if it makes a different on this specific test, though.